### PR TITLE
Update ::sapi and move to um::sapi

### PR DESCRIPTION
**Pull request by docbrown**
Work in progress. I still need to clear up a few things in `um::sapi` and add `um::sapiaut` + `um::sapiddk`. This PR also brings in `um::servprov` and an incomplete `um::urlmon`. I can either finish the latter in this PR or in a separate one later.

### CODE REVIEW

**Comment by docbrown** [comment]
On second thought, `um::sapiaut` can probably be omitted because it doesn't have a header file. It seems to be intended as a typelib only.

**Comment by retep998** [comment]
The stuff from `sapiaut.idl` looks like it is actually defined in `sapi51.h` and `sapi53.h`.

**Comment by docbrown** [comment]
@retep998 Yep, you're right.

File: src/um/sapiddk.rs [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+use um::sapi::*;
+use um::unknwnbase::{IUnknown, IUnknownVtbl};
+use um::winnt::{HRESULT, LPCWSTR, ULONGLONG, WCHAR};
+pub use um::sapiddk51::{
+    SPRECOEXTENSION,
+    SPALTERNATESCLSID,
+};
+pub const SR_LOCALIZED_DESCRIPTION: &'static str = "Description";
+extern {
+    pub static __MIDL_itf_sapiddk_0000_0000_v0_0_c_ifspec: RPC_IF_HANDLE;
```
Comment (retep998) [inline src/um/sapiddk.rs]: These should just be left out. They're stuff leftover by IDL and don't seem to be useful at all.

**Comment by docbrown** [comment]
I still have a few things to sort out, plus everything from sapiaut.idl needs to be added. Should be done tomorrow.

**Comment by docbrown** [comment]
Nearly there. Just need to figure out `ISpNotifyCallback` (maybe add a UUID-less pattern to `RIDL!`?) and add a `BITFIELD!` for `SPEVENT`.

**Comment by docbrown** [comment]
That should do it!

**Comment by retep998** [comment]
PR has conflicts.

**Comment by docbrown** [comment]
I still need to add trailing commas to interface methods.

**Comment by Susurrus** [comment]
@docbrown Any interest in wrapping up this PR? I'm now blocking on some datatypes in `sapi` for porting more functionality in `winapi` to 0.3.

**Comment by retep998** [comment]
@GuillaumeGomez This PR is failing at your tests.

**Comment by GuillaumeGomez** [comment]
@retep998: Yes I know, I fixed my tests in #440.

**Comment by docbrown** [comment]
The failures are due to re-exports not satisfying the check_imports test. Once it's fixed, this PR should be ready to go.

**Comment by Susurrus** [comment]
@docbrown Thanks. Yeah it looks like once #440 is merged, this can have its tests re-run and be ready for a merge. Great!

**Comment by Susurrus** [comment]
@docbrown Can you force-force this PR again to re-run the tests. #440 was merged so this should be good to go.

**Comment by Susurrus** [comment]
@docbrown Tests are still failing, looks like your library includes aren't correct.

**Comment by Susurrus** [comment]
LGTM now, thanks @docbrown for fixing the tests as well. Now just waiting for @retep998 to review and merge.

**Pull request merged**